extension.js

```diff
@@ -173,6 +173,33 @@ const ExtensionLayout = new Lang.Class({
     GLib.spawn_command_line_async(cmd);
   },
 
+  _findNewStreamerEntries:function(lastList, currentList, detectGameChange) {
+    detectGameChange = detectGameChange || false;
+    if ( lastList.length == 0 )
+      return currentList;
+
+    let newEntries = [];
+    // filter out all lastList items from currentList
+    currentList.forEach(entry => {
+      let match = false;
+
+      // Use a for loop so we can break it off when we match
+      for ( var i = 0; i < lastList.length; i++ ) {
+        let last = lastList[i];
+        if ( entry.streamer == last.streamer )
+          if ( !detectGameChange || ( detectGameChange && entry.game == last.game ) ) {
+            match = true;
```

Comment:
I'm not entirely sure what my rationale was for doing this match/break instead of arr.push()-ing right here.

Reply:
oooh, I was using more than one if/else branch so this made sense.